src/expression.d

```diff
@@ -14159,14 +14159,19 @@ extern (C++) final class MinExp : BinExp
         {
             if (t2.ty == Tpointer)
             {
+                Type p1 = t1.nextOf();
+                Type p2 = t2.nextOf();
+
+                if (!p1.equivalent(p2))
```

Comment:
I'd expect this to break a lot of code, maybe rather require types of same size than requiring identity?